src/MongoDB.Driver/Encryption/AutoEncryptionLibMongoController.cs

```diff
@@ -42,7 +44,7 @@ internal sealed class AutoEncryptionLibMongoCryptController : LibMongoCryptContr
         {
             _client = Ensure.IsNotNull(client, nameof(client)); // _client might not be fully constructed at this point, don't call any instance methods on it yet
             _mongocryptdFactory = new MongocryptdFactory(autoEncryptionOptions.ExtraOptions);
-            _mongocryptdClient = _mongocryptdFactory.CreateMongocryptdClient();
+            _mongocryptdClient = Ensure.IsNotNull(_mongocryptdFactory.CreateMongocryptdClient(), nameof(_mongocryptdClient));
```

Comment:
EnsureIsNotNull is used to test argument validity, not intermediate results.

Why would CreateMongocryptdClient ever return null? It should either return a client or throw an exception if it can't create one.